students/mkdir01/lesson04/mailroom2.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+import sys
+
+# changed from list of lists to a dict with values as list
+data = {'John Doe': [120.00, 353.33, 400.00], 'Jane Doe': [3500.04, 2624.44], 'John Smith': [33.33, 400.00, 29.20], 'Jane Smith': [2.00], 'Billy Jo Jones': [100.00, 100.00, 100.00]}
+
+
+# mode 1 - "send a thank you"
+def thank_you():
+    while True:
+        name = input("Please enter a full name (or type \'list\', or hit \'return\' to go back)> ")
+        if name == "list":
+            for k in data:
+                print(k)
+        elif name == "":
+            return
+        else:
+            dollars = input("Please enter a donation amount (ex: 500.05)> ")
+            for k in data:
+                if k == name:
+                    data[k].append(round(float(dollars), 2))
+                    break
```

Comment:
looks great! to make this even easier I would recommend taking a look at defaultdict from collections module:
https://www.accelebrate.com/blog/using-defaultdict-python/